src/Blubrry/REST/Resource/Social.php

```diff
@@ -79,6 +177,44 @@ public function postSocial($program_keyword, $body) {
             }
         }
 
+        $type=$body['social-type'];
+        $twitter=array('content');
+
+        $facebook=array(
+            "title-$body['social-id']",
+            "description-$body['social-id']",
+            "destination-$body['social-id']",
+        );
+
+        $youtube=array(
+            "title-$body['social-id']",
+            "description-$body['social-id']",
+        );
+
+        if ($type == "twitter") {
+            foreach ($twitter as $item) {
+                if (is_null($body['social-data'][$item])) {
+                    return false;
+                }
+            }
+        }
+
+        if ($type == "facebook") {
+            foreach ($facebook as $item) {
+                if (is_null($body['social-data'][$item])) {
+                    return false;
+                }
+            }
+        }
+
+        if ($type == "youtube") {
+            foreach ($youtube as $item) {
```

Comment:
@lucashillebrandt Do you really need to have 3 different `for` to do the same thing?

Couldn't you simplify it by doing something like the below?

```php
if (preg_match('/(facebook|youtube|twitter)/', $type)) {
```